proto/spire/agent/svidstore/svidstore.proto

```diff
@@ -0,0 +1,49 @@
+syntax = "proto3";
+package spire.agent.svidstore;
+option go_package = "github.com/spiffe/spire/proto/spire/agent/svidstore";
+
+import "spire/common/common.proto";
+import "spire/common/plugin/plugin.proto";
+
+service SVIDStore {
+    // Puts an X509-SVID in a configured secrets store
+    rpc PutX509SVID(PutX509SVIDRequest) returns (PutX509SVIDResponse);
+
+    // Applies the plugin configuration and returns configuration errors
+    rpc Configure(spire.common.plugin.ConfigureRequest) returns (spire.common.plugin.ConfigureResponse);
+
+    // Returns the version and related metadata of the plugin 
+    rpc GetPluginInfo(spire.common.plugin.GetPluginInfoRequest) returns (spire.common.plugin.GetPluginInfoResponse);
+}
+
+message PutX509SVIDRequest {
+    // X509-SVID to be stored
+    X509SVID svid = 1;
+
+    // Entry selectors with relevant information to store on specific platform
+    repeated spire.common.Selector selectors = 2;
```

Comment:
I use common selectors because our cache maintain selectors using it, `but` may we use `spire.types.Selector` (https://github.com/spiffe/spire/blob/master/proto/spire/types/selector.proto#L5) instead?
At least in this proto and in a future we may update our cache to use new protos, with that we'll void parsing protos we get from SPIRE Server API.
@azdagron @evan2645 @amartinezfayo @rturner3  any ideas about it?

Reply:
If we want to use `spire.types.Selector` rather than `spire.common.Selectors` in the Agent cache, we'd probably have to either:

1. Make a breaking change to the `WorkloadAttestor` Agent plugin interface to use the type from `spire.types` and update all the Agent code to use the `spire.types` version

or

2. Translate instances of `spire.common.Selector` in Agent code to `spire.types.Selector` in the WorkloadAttestor handler and use `spire.types.Selector` everywhere else in the Agent code

Either option seems feasible to me, although with option 1, we would have to wait till a major release to break the `WorkloadAttestor` contract.

Reply:
mmm, make sense, I'm just worried about using `common` selector in a new proto,
I dont know if it makes sense to use `types` selectors here, we can create them from common selectors before calling `PutX509SVID`,
but we can update it after we update cache to use new `types` in a future of course.

Reply:
Just a drive by comment here :) The `types` package was introduced for the APIs. In the past there has not been a clear line between API related messages and plugin related messages. Coupling them together, while attractive from a DRY perspective, isn't really attractive from a "these things should be able to grow independently" perspective.

Considering all the other plugins still rely on the common package, I'd be hesitant to go one-off just for this plugin. IMO, we should continue to use the common package for plugins. At some point in time we can make a change to types if we want to, but now doesn't seem to be the right time.